Framework/PythonInterface/mantid/api/src/PythonAlgorithm/AlgorithmAdapter.cpp

```diff
@@ -334,7 +334,14 @@ template <typename BaseAlgorithm> void AlgorithmAdapter<BaseAlgorithm>::init() {
  * overridden in the subclass by a function named PyExec
  */
 template <typename BaseAlgorithm> void AlgorithmAdapter<BaseAlgorithm>::exec() {
-  callMethod<void>(getSelf(), "PyExec");
+  try {
+    callMethod<void>(getSelf(), "PyExec");
+  } catch (Mantid::PythonInterface::PythonException &) {
+    if (BaseAlgorithm::getCancel())
+      throw Mantid::API::Algorithm::CancelException();
```

Comment:
This is indeed forces Algorithm to catch the correct exception. How do you distinguish between an actual error state vs cancellation in the Python code or do you not need to do this currently?

Reply:
@martyngigg Currently the existing exception translations in the Algorithm.cpp export file change a CancelException into a KeyboardInteruptException. https://github.com/mantidproject/mantid/blob/master/Framework/PythonInterface/mantid/api/src/Exports/Algorithm.cpp#L86

This is enough to distinguish for now between actual error states and cancellations in the python code particularly as KeyBoardInterupt inherits from BaseException and not Exception. In an ideal world we would check for the CancellationException explicitly though.

Reply:
Ah I see. I think I missed where the `KeyboardInterrupt` was being used. You are correct in that I think for long term we should actually not use `KeyboardInterrupt` at all as it's generally considered bad practice to catch `BaseException` types that don't inherit `Excecption` except in really low level code.

Could quickly check that this doesn't impede the script window's abort behaviour? That uses `KeyboardInterrupt` to signal it to stop and if we just catch it then this may not work.

Reply:
sorry I dont understand what you mean for the test. Is it to run a script and abort it while the frequency GUI is open?

Reply:
Ah actually I think I may have misunderstood something here. This class is entirely for the GUI? My concern was around catching a `KeyboardInterrupt` and not re-raising it as this would impede some abort operations. If this is only used in the GUI I think that is okay though.